src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/integration/endpoint/ExternalDocRequestEndpointTest.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.digital.hmpps.crimeportalgateway.integration.endpoint
+
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.ApplicationContext
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
+import org.springframework.ws.test.server.MockWebServiceClient
+import org.springframework.ws.test.server.RequestCreators
+import org.springframework.xml.transform.StringSource
+import uk.gov.justice.digital.hmpps.crimeportalgateway.config.WebServiceConfig
+import javax.xml.transform.Source
+
+//@RunWith(SpringJUnit4ClassRunner::class)
+@SpringBootTest
+@ContextConfiguration(classes = [WebServiceConfig::class])
+internal class ExternalDocRequestEndpointTest {
+
+    // https://docs.spring.io/spring-ws/docs/current/reference/#_server_side_testing
+
+    @Autowired
+    private val applicationContext: ApplicationContext? = null
+
+    private var mockClient: MockWebServiceClient? = null
+
+    @Before
+    fun before() {
+        mockClient = MockWebServiceClient.createClient(applicationContext)
+    }
+
+    @Ignore
+    @Test
+    fun `should return SOAP header and body`() {
```

Comment:
Put a breakpoint in the endpoint and it isn't being found. Probably the initialisation of the spring boot app.

Reply:
I'll have a look at this tomorrow if that's ok so I can give it my full attention. Happy to merge it in for now and we can remove it if we can't get it working.